core/src/main/scala/org/apache/spark/util/Clock.scala

```diff
@@ -99,3 +101,20 @@ private[spark] class SystemClock extends Clock {
     -1
   }
 }
+
+/**
+ * To return a mocked system clock for testing purposes
+ */
+private[spark] class MockedSystemClock extends ManualClock {
+  override def getTimeMillis(): Long = {
+    currentMockSystemTime
+  }
+}
+
+object MockedSystemClock {
```

Comment:
even though trivial, I would suggest having the habit of adding a top level comment on classes and objects. How about:

Allows advancing the global static time of ManualSystemClock (or whatever class name you prefer)

Reply:
This is a companion object of MockedSystemClock.  It basically contains static methods for that class.  I think having a comment for the MockedSystemClock is good enough